src/domUpdates.js

```diff
@@ -42,9 +45,66 @@ let domUpdates = {
     })
   },
 
-  displayShoppingList(ingredientsNeeded) {
-    makeRecipeMessage.innerText = `You do not have enough ingredients in your pantry to make this recipe.`
-    // shopping list functionality shows up here... add button so you can click to expand?
+  displayShoppingList(shoppingList) {
+    modalShoppingMessageYes.style.display = 'none'
+    modalShoppingList.style.display = 'block'
+    shoppingList.forEach(shoppingItem => {
+      const listItem = document.createElement('tr')
+      const itemName = document.createElement('td')
+      const itemQuantity = document.createElement('td')
+      const itemCostPerUnit = document.createElement('td')
+      const itemTotalCost = document.createElement('td')
+      let unit = this.formatUnits(shoppingItem.unit)
+      let totalCost = (shoppingItem.cost * shoppingItem.quantity)/100
+
+      itemName.innerText = shoppingItem.name
+      itemQuantity.innerText = `${shoppingItem.quantity} ${unit}`
+      itemCostPerUnit.innerText = `$${shoppingItem.cost/100}`
+      itemTotalCost.innerText = `$${totalCost.toFixed(2)}`
+
+      itemCostPerUnit.classList.add("price")
+      itemTotalCost.classList.add("price")
+
+      modalShoppingItems.appendChild(listItem)
+      listItem.appendChild(itemName)
+      listItem.appendChild(itemQuantity)
+      listItem.appendChild(itemCostPerUnit)
+      listItem.appendChild(itemTotalCost)
+    })
+
+    this.displayTotalCost(shoppingList)
+  },
+
+  displayTotalCost(shoppingList) {
+    const totalListCost = shoppingList.reduce((totalCost, item) => {
+      totalCost += (item.cost * item.quantity) / 100
+      return totalCost
+    }, 0)
+
+    modalTotalCost.innerHTML = `
+        <tr>
+          <td colspan="4" class="price">Total Cost: $${totalListCost.toFixed(2)}</td>
+        <tr>`
+  },
+
+  formatUnits(unit) {
+    if (unit === "teaspoons" || unit === "teaspoon") {
```

Comment:
It might be a good idea to assign that `unit` parameter value to a variable as `unit.toLowerCase()` or `toUpperCase()`. That won't change much considering you only have to account for case in one `if` statement, but I think its a more robust approach and will do a little but as far as readability goes.

Reply:
Ooh I think I see what you mean. Let's look at this one together in stand up.